src/main.rs

```diff
@@ -11,21 +11,21 @@ fn main() {
     }
 }
 
-trait Integrator {
-    fn step<F: Fn(f64, f64) -> f64>(&self, f: &F, x0: f64, t0: f64, dt: f64) -> f64;
+trait Integrator<F: Mul<f64, Output = F> + Add<Output = F> + Copy> {
+    fn step(&self, f: &Fn(f64, F) -> F, x0: F, t0: f64, dt: f64) -> F;
```

Comment:
I'm not sure we want to do this as a reference to a trait object instead of a generic parameter, this likely prevents inlining the function into the integrator.

Reply:
Since we're stealing the `F` that we used for functions, maybe we should rename the vector type parameter `X`?